lstchain/datachecks/containers.py

```diff
@@ -0,0 +1,371 @@
+                               'consecutive events')
+    hist_npixels = Field(None, 'Histogram of number of pixels in image')
+    hist_nislands = Field(None, 'Histogram of number of islands in image')
+    hist_intensity = Field(None, 'Histogram of image intensity')
+    hist_dist0 = Field(None, 'Histogram of cog-camera center distance')
+    hist_dist0_intensity_gt_200 = \
+        Field(None, 'Histogram of cog-camera center distance')
+    hist_width = Field(None, 'Histogram image width vs. intensity')
+    hist_length = Field(None, 'Histogram image length vs. intensity')
+    hist_skewness = Field(None, 'Histogram of image skewness')
+    # the histogram hist_pixelchargespectrum shows the pixel charge
+    # distribution, filled from all pixels:
+    hist_pixelchargespectrum = Field(None, 'Histogram of pixel charges')
+
+    hist_psi = Field(None, 'Histogram of image axis orientation')
+    hist_intercept = Field(None, 'Histogram of fitted pulse time for charge '
+                                 'c.o.g.')
+    hist_tgrad_vs_length = Field(None, 'Histogram of time gradient vs. length')
+    hist_tgrad_vs_length_intensity_gt_200 = \
+        Field(None, 'Histogram of time gradient vs. length, intensity>200pe')
+
+    # pixel-wise quantities, one entry per pixel. Used also for 2d
+    # histogramming of cog position.
+    cog_within_pixel = Field(None, 'Number of image cogs within pixel')
+    cog_within_pixel_intensity_gt_200 = \
+        Field(None, 'Number of image within pixel, intensity>200pe')
+    charge_mean = Field(-1, 'Mean of pixel charge')
+    charge_stddev = Field(-1, 'Standard deviation of pixel charge')
+    time_mean = Field(-1, 'Mean of pulse time')
+    time_stddev = Field(-1, 'Standard deviaton of pulse time')
+    time_mean_above_030_pe = Field(-1, 'Mean of pulse time, >30 p.e. pulses')
+    time_stddev_above_030_pe = Field(-1, 'Standard deviaton of pulse time, '
+                                         '>30 p.e. pulses')
+    relative_time_mean = Field(-1, 'Mean of pulse time relative to average of '
+                                   'rest of pixels')
+    relative_time_stddev = Field(-1, 'Standard deviaton of pulse time '
+                                     'relative to average of rest of pixels')
+
+    # keep number of events above a few thresholds, like a low-res histogram
+    # of pulse charges (2 points per decade in charge in p.e.)
+    # This could be done in a cleaner way with a 2d hist charge vs. pixel (TBD)
+    num_pulses_above_0010_pe = Field(None, 'Number of >10 p.e. pulses')
+    num_pulses_above_0030_pe = Field(None, 'Number of >30 p.e. pulses')
+    num_pulses_above_0100_pe = Field(None, 'Number of >100 p.e. pulses')
+    num_pulses_above_0300_pe = Field(None, 'Number of >300 p.e. pulses')
+    num_pulses_above_1000_pe = Field(None, 'Number of >1000 p.e. pulses')
+
+    def fill_event_wise_info(self, subrun_index, table, mask, geom,
```

Comment:
This is a more general comment for all of lstchain. BUT: in ctapipe containers are just that, containers that hold data. 

Here you add methods to these, that do a lot of things. Why? Why not write a free function that creates and returns a container?

I think containers should be kept simple and not have a plethora of methods.

Reply:
> This is a more general comment for all of lstchain. BUT: in ctapipe containers are just that, containers that hold data.
> 
> Here you add methods to these, that do a lot of things. Why? Why not write a free function that creates and returns a container?
> 
> I think containers should be kept simple and not have a plethora of methods.

They are mostly filling methods that are specific for that type of container, so I do not see an advantage in dettaching them from the container class. I see no advantage in them being methods rather than functions either, so we can discuss it for the future.

Reply:
It's just a clear break with how ctapipe uses these containers and there is no obvious reason for me to do this

Reply:
> It's just a clear break with how ctapipe uses these containers and there is no obvious reason for me to do this

Ok, we can address this when we do other ctapipe-adaptations to start using the future ctapipe 0.8 release